src/configuration-requestor.spec.ts

```diff
@@ -111,105 +115,217 @@ describe('ConfigurationRequestor', () => {
+      });
 
-    it('Will not fetch bandit parameters if there is no store', async () => {
-      configurationRequestor = new ConfigurationRequestor(httpClient, flagStore, null, null);
-      await configurationRequestor.fetchAndStoreConfigurations();
-      expect(fetchSpy).toHaveBeenCalledTimes(1);
-    });
+      it('Should not fetch bandits if model version is un-changed', async () => {
+        await configurationRequestor.fetchAndStoreConfigurations();
+        expect(fetchSpy).toHaveBeenCalledTimes(2); // Once for UFC, another for bandits
 
-    it('Requests bandits only when model versions are different', async () => {
-      await configurationRequestor.fetchAndStoreConfigurations();
-      expect(fetchSpy).toHaveBeenCalledTimes(2); // Once for UFC, another for bandits
+        await configurationRequestor.fetchAndStoreConfigurations();
+        expect(fetchSpy).toHaveBeenCalledTimes(3); // Once just for UFC, bandits should be skipped
+      });
 
-      await configurationRequestor.fetchAndStoreConfigurations();
-      expect(fetchSpy).toHaveBeenCalledTimes(3); // Once just for UFC, bandits should be skipped
+      it('Should fetch bandits if new bandit references model versions appeared', async () => {
+        let updateUFC = false;
+        await configurationRequestor.fetchAndStoreConfigurations();
+        await configurationRequestor.fetchAndStoreConfigurations();
+        expect(fetchSpy).toHaveBeenCalledTimes(3);
+
+        const customResponseMockGenerator = (url: string) => {
+          const responseFile = url.includes('bandits')
+            ? MOCK_BANDIT_MODELS_RESPONSE_FILE
+            : MOCK_FLAGS_WITH_BANDITS_RESPONSE_FILE;
+
+          const response = readMockUFCResponse(responseFile);
+
+          if (updateUFC === true) {
+            // this if is needed to appease linter
+            if (url.includes('config') && 'banditReferences' in response) {
+              response.banditReferences.warm_start = {
+                modelVersion: 'warm start',
+                flagVariations: [
+                  {
+                    key: 'warm_start_bandit',
+                    flagKey: 'warm_start_bandit_flag',
+                    variationKey: 'warm_start_bandit',
+                    variationValue: 'warm_start_bandit',
+                  },
+                ],
+              };
+            }
+
+            if (url.includes('bandits') && 'bandits' in response) {
+              response.bandits.warm_start = {
+                banditKey: 'warm_start_bandit',
+                modelName: 'pigeon',
+                modelVersion: 'warm start',
+                modelData: {
+                  gamma: 1.0,
+                  defaultActionScore: 0.0,
+                  actionProbabilityFloor: 0.0,
+                  coefficients: {},
+                },
+              };
+            }
+          }
+          return response;
+        };
+        updateUFC = true;
+        initiateFetchSpy(customResponseMockGenerator);
```

Comment:
e.g., you should do another fetch before and make sure bandits is not fetched--ideally with a _subset_